compiler/ast/reports.nim

```diff
@@ -410,11 +411,18 @@ func reportSym*(
 
   SemReport(kind: kind, ast: ast, str: str, typ: typ, sym: sym)
 
-template withIt*(expr: untyped, body: untyped): untyped =
-  block:
-    var it {.inject.} = expr
-    body
-    it
+# template withIt*(expr: untyped, body: untyped): untyped =
+#   block:
+#     var it {.inject.} = expr
+#     stripDoNode(body)
+#     it
+macro withIt*(expr: untyped, body: untyped): untyped =
```

Comment:
Not sure if changing to `do` section is necessary - it worked perfectly fine without that, at least of the use cases I had. The template is supposed to emulate rust's field update uses.

Reply:
it was used somewhere with an explicit `do` argument and that do node needs to be taken out again. I wanted to change the usage to not use `do` but it was weird.